### ISSUE TITLE
isDH in newPop() can be confusing

### ISSUE THREAD

**Issue by gregorgorjanc** [gaynorr/AlphaSimR#21]
newPop() docs say that "isDH - optional value indicating if the individuals are doubled haploids and/or inbred founders". Some users might think that this can double genomes. We should reword the docs to make it even clearer that this is only an indication of the DH state and that it does not create DHs.

```
> library(AlphaSimR)
> founderGenomes = runMacs(nInd = 2,
+                          nChr = 1,
+                          segSites = 10,
+                          species = "MAIZE")
> SP = SimParam$new(founderGenomes)
> pullSegSiteHaplo(newPop(founderGenomes))
    SITE_1 SITE_2 SITE_3 SITE_4 SITE_5 SITE_6 SITE_7 SITE_8 SITE_9 SITE_10
1_1      1      0      0      0      1      0      1      0      1       1
1_2      1      0      0      1      0      1      0      0      0       1
2_1      1      0      1      1      0      1      1      0      0       1
2_2      0      1      0      1      1      0      1      1      0       0
> pullSegSiteHaplo(newPop(founderGenomes, isDH = TRUE))
    SITE_1 SITE_2 SITE_3 SITE_4 SITE_5 SITE_6 SITE_7 SITE_8 SITE_9 SITE_10
3_1      1      0      0      0      1      0      1      0      1       1
3_2      1      0      0      1      0      1      0      0      0       1
4_1      1      0      1      1      0      1      1      0      0       1
4_2      0      1      0      1      1      0      1      1      0       0

> sessionInfo()
R version 4.0.4 (2021-02-15)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] mixtools_1.2.0   AlphaSimR_0.13.0 R6_2.5.0        

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.6        lattice_0.20-41   digest_0.6.27     crayon_1.4.1      MASS_7.3-53      
 [6] prettycode_1.1.0  grid_4.0.4        pacman_0.5.1      evaluate_0.14     rlang_0.4.10     
[11] kernlab_0.9-29    Matrix_1.3-2      fortunes_1.5-4    rmarkdown_2.7     splines_4.0.4    
[16] tools_4.0.4       survival_3.2-7    xfun_0.22         yaml_2.2.1        compiler_4.0.4   
[21] segmented_1.3-4   htmltools_0.5.1.1 knitr_1.31       
```

**Comment by gregorgorjanc** [comment]
Or, call `makeDH()` when that flag is set to TRUE?!

**Comment by gaynorr** [comment]
I've decided to "hide" some of this from the user. It was intended for passing a flag to the pedigree tracking functionality and not designed for actually doing anything to the population. I can see how this isn't very clear for the documentation. It's also probably very unclear how the function plays two roles in simulations. The first is to initialize the simulation, something the user does directly. The second is to create all subsequent populations, something that happens without the user's direct input. 

I've decided to deal with this by creating two versions of the function. The first is called .newPop. It that retains the functionality of the original function and is not exported to the package namespace. The second retains the name of the original, but is only designed as a wrapper for the first function that gets used by users to initialize the simulation. The new newPop function can pass along many additional variables to .newPop via the "..." argument.
